src/app/inventory/d2-season-info.ts

```diff
@@ -73,6 +85,33 @@ export const D2SeasonInfo = {
     year: 2,
     maxLevel: 50,
     maxPower: 750,
-    softCap: 500
+    softCap: 500,
+    release: 'June 25 2019', // TODO: Update this upon confirmation
+    resetTime: '17:00 UTC'
   }
 };
+
+function enumLength(enumName: any): number {
+  // https://stackoverflow.com/questions/38034673/determine-the-number-of-enum-elements-typescript
+  let count = 0;
+  for (const item in enumName) {
+    if (isNaN(Number(item))) {
+      count++;
+    }
+  }
+  return count;
+}
+
+function getCurrentSeason(): number {
+  let seasonDate: Date;
+  const today = new Date(Date.now());
+  for (let i = enumLength(D2SeasonEnum); i--; ) {
+    seasonDate = new Date(`${D2SeasonInfo[i].release} ${D2SeasonInfo[i].resetTime}`);
```

Comment:
Does this work in all browser locales?

Reply:
According to https://flaviocopes.com/javascript-dates/ it should... Maybe @48klocs would have better insight.